scripts/test/Muon/elemental_analysis_2/elemental_analysis_load_widget_presenter_test.py

```diff
@@ -0,0 +1,76 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from unittest import mock
+
+from mantidqt.utils.qt.testing import start_qapplication
+from qtpy.QtWidgets import QWidget
+
+from Muon.GUI.Common.load_file_widget.view import BrowseFileWidgetView
+from Muon.GUI.Common.load_run_widget.load_run_view import LoadRunWidgetView
+from Muon.GUI.Common.test_helpers.context_setup import setup_context_for_ea_tests
+from Muon.GUI.ElementalAnalysis2.load_widget.load_models import LoadWidgetModel, LoadRunWidgetModel
+from Muon.GUI.ElementalAnalysis2.load_widget.load_widget_presenter import LoadWidgetPresenterEA
+from Muon.GUI.ElementalAnalysis2.load_widget.load_run_widget_presenter import LoadRunWidgetPresenterEA
+from Muon.GUI.MuonAnalysis.load_widget.load_widget_view import LoadWidgetView
+
+
+@start_qapplication
+class LoadWidgetPresenterTest(unittest.TestCase):
+    def setUp(self):
+        # Store an empty widget to parent all the views, and ensure they are deleted correctly
+        self.obj = QWidget()
+
+        setup_context_for_ea_tests(self)
+        self.load_file_view = BrowseFileWidgetView(self.obj)
+        self.load_run_view = LoadRunWidgetView(self.obj)
+        self.load_run_view.set_current_instrument = mock.Mock()
+        self.view = LoadWidgetView(parent=self.obj,
+                                   load_file_view=self.load_file_view,
+                                   load_run_view=self.load_run_view)
+        self.model = LoadWidgetModel(self.loaded_data, self.context)
+        self.model.instrument = mock.Mock()
+        self.presenter = LoadWidgetPresenterEA(self.view, self.model)
+        loaded_data = mock.Mock()
+        context = mock.Mock()
+        self.run_widget = LoadRunWidgetPresenterEA(self.load_run_view, LoadRunWidgetModel(loaded_data, context))
+        self.presenter.set_load_run_widget(self.run_widget)
+
+    def tearDown(self):
+        self.obj = None
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def test_model_initialized_to_contain_no_data(self):
+        self.assertEqual(self.model.runs, [])
+
+    def test_clear_data(self):
+        self.model.current_runs = [[1234], [1235]]
```

Comment:
this works. However, you could mock the model and then check that the presenter calls it with the correct arguments. You know what the model will from the model tests.